ports/esp8266/modnetwork.c

```diff
@@ -93,29 +93,55 @@ STATIC mp_obj_t esp_active(size_t n_args, const mp_obj_t *args) {
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(esp_active_obj, 1, 2, esp_active);
 
-STATIC mp_obj_t esp_connect(size_t n_args, const mp_obj_t *args) {
-    require_if(args[0], STATION_IF);
+STATIC mp_obj_t esp_connect(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    enum { ARG_essid, ARG_password, ARG_bssid };
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_essid, MP_ARG_OBJ, {.u_obj = mp_const_none} },
```

Comment:
As discussed in the ticket, these aren't expected to be passed by kwarg, only positionally. So, can these be set as MP_QSTR_ instead?

Reply:
But this is a specific NIC (not an AbstractNIC) and so it's possible to give the parameters specific names.  The qstrs already exist elsewhere so it doesn't add anything to the code size and gives users a better experience.  What's the concern having them here?

Reply:
Well, if these qstr's already exist, there's less concern. A background concern though is being compliant with the word of spec, and provide a model code for other ports to follow. Or change the spec, if there's no desire to follow it and instead workarounds being applied.

 > better experience

Do you mean possible use of param name in error message (I don't even remember if we do it), or ability to pass ssid/password by kwargs? In the latter case, the spec was specifically designed to avoid that, to have an API where 1st are is something identifying service, and 2nd - a key to access it.

Reply:
Error messages do name the parameter that's missing, but only if it's a required argument, which in this case it's not.  So I guess it is ok not to name these parameters, it won't lead to any strange behaviour.